### PR TITLE
Add Line Mapping Support to Hook

### PR DESCRIPTION
**Pull request by fabiosvm**
Add line mapping support to Hook language.
Update Hook's logo.

### CODE REVIEW

**Review by RubenRBS — APPROVED** [review]

File: lib/compilers/hook.ts [hunk 1]
```diff
@@ -52,4 +54,33 @@ export class HookCompiler extends BaseCompiler {
         options.push(outputFilename);
         return super.runCompiler(compiler, options, inputFilename, execOptions);
     }
+
+    override processAsm(result) {
```
Comment (RubenRBS) [inline lib/compilers/hook.ts:58]: Would appreciate it if you could add some tests for this :)

Comment (fabiosvm) [reply lib/compilers/hook.ts:58]: Sure. I'll do.

Comment (fabiosvm) [reply lib/compilers/hook.ts:58]: Are the compiler tests in that `test/compilers` folder? I just found a `clang-tests.js` in there.

Comment (RubenRBS) [reply lib/compilers/hook.ts:58]: Not much testing is done for the compilers themselves which is what we're trying to change, yes :) But feel free to create the tests wherever makes more sense to you

Comment (fabiosvm) [reply lib/compilers/hook.ts:58]: @RubenRBS, I covered the lines that this PR touched on. See if that's enough, please.

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**